### PR TITLE
Fixes pretty printing for uri parser

### LINKED ISSUE #1478: parse failure error for git urls needs improvement
Currently:
```
.base> pull https://github.com:unisonweb/base:.trunk

  TrivialError (SourcePos {sourceName = "url", sourceLine = Pos 1, sourceColumn = Pos 20} :| []) (Just (Tokens ('u' :| ""))) (fromList [Label ('d' :| "igits")])

.base>
```

Proposed:
```
.base> pull https://github.com:unisonweb/base:.trunk

  I couldn't parse the repository address given above.

    https://github.com:unisonweb/base:.trunk
                       ^-- This is where I gave up.

.base>
```

### PR DESCRIPTION
**Pull request by solomon-b**
## Overview

Fixes #1478 

Fixes the uri parser pretty printing:

```
.> pull https://github.com:unisonweb/base:.trunk:

  I couldn't parse the repository address given above.
  
  
  https://github.com:unisonweb/base:.trunk:
                     ^-- This is where I gave up.

```

## Implementation notes

The custom error type was `()` but it should be `Void`. I switched that.

## Interesting/controversial decisions

Nothing I can think of?

## Test coverage

I didn't see any unhappy path parser tests so I wasn't sure how I should add any.

## Loose ends

Link to related issues that address things you didn't get to. Stuff you encountered on the way and decided not to include in this PR.

### CODE REVIEW

File: unison-cli/src/Unison/CommandLine/InputPatterns.hs [hunk 1]
```diff
@@ -1165,8 +1167,26 @@ loadPullRequest =
...
+      in P.lines ["I couldn't parse the repository address given above.",
+               P.newline,
+               P.string errorLine,
+               P.string $ replicate col ' ' <> "^-- This is where I gave up."
+              ]
+    prettyPrint (P.FancyError sp _) =
+      let col = (P.unPos $ P.sourceColumn $ NE.head sp) - 1
+      in P.lines ["I couldn't parse the repository address given above.",
+               P.newline,
+               P.string input,
```
Comment (ChrisPenner) [inline unison-cli/src/Unison/CommandLine/InputPatterns.hs]: I notice in the TrivialError case you need to grab the appropriate row, do we not need to do that for FancyErrors too?
I suspect we'll usually be parsing URI's independently as a single line, but we should still handle that case if we can.

It seems possible to unify these two implementations into a single one which just takes the span right? Or am I missing something there?

**Review by ChrisPenner — APPROVED** [review]
Just left a note about the two different implementations, maybe we can unify them.

Also, is it possible to get megaparsec's message about "I was expecting _ but got _" in there somewhere? It would be nice to know what it was expecting in addition to _where_ it was expecting it.

Even if that's not possible, what you've got here is already much better than we had before 👍🏼

**Comment by solomon-b** [comment]
I'll take a look at this today and see if I can improve the implementation!

**Comment by ChrisPenner** [comment]
@solomon-b  if it ends up being tricky to apply the feedback I'd be happy to merge this as-is, just let me know 👍🏼

**Comment by solomon-b** [comment]
@ChrisPenner Sorry I started updating it but got sidetracked. I'll push a change today or tomorrow!

**Comment by solomon-b** [comment]
@ChrisPenner ok i just did a cleanup on the pretty printer. What do you think?

In the trivial error case we get:
```
.> pull https://github.com:unisonweb/base:.trunk:

  I couldn't parse the repository address given above.
  
  
  https://github.com:unisonweb/base:.trunk:
                     ^-- This is where I gave up.
  
  
  I expected 'digits' and I found 'u'.
```
I don't like the way the expected token is printed but that comes from the `ShowErrorComponent` instance in Megaparsec. I don't like that you can't distinguish if it is showing a list of expected tokens or a label for a type of token (i think this depends on the parser).

**Comment by ChrisPenner** [comment]
Looks good! Thanks Solomon 😄

**Pull request merged**